src/HotChocolate/Core/src/Types/Types/InputObjectType.cs

```diff
@@ -133,19 +133,9 @@ public virtual bool TrySerialize(object value, out object serialized)
 
         public object Deserialize(object serialized)
         {
-            if (serialized is null)
+            if (TryDeserialize(serialized, out object deserialized))
             {
-                return null;
-            }
-
-            if (serialized is IReadOnlyDictionary<string, object> dict)
-            {
-                return _deserialize(dict);
```

Comment:
is this method still used elsewhere or can it also be removed?

Reply:
you mean the Deserialize method?

Reply:
not the method, but the `private Func<IReadOnlyDictionary<string, object>, object> _deserialize`